### DISCUSSION TITLE
Problems upgrading server to r3_8_2

### DISCUSSION THREAD

**Discussion by NickVeit** [jamulussoftware/jamulus#2416] [Technical Support & Help] [answered]
My Ubuntu knowledge is zero. I usually upgrade my server using this:

cd /usr/local/bin/llcon-jamulus
sudo git pull
sudo git checkout r3_8_1
sudo qmake "CONFIG+=nosound" Jamulus.pro
sudo make
sudo systemctl restart jamulus

When I checkout r3_8_1 I get "HEAD is now at 40fd0ac7 Update version to 3.8.1 for release"

When I try to checkout r3_8_2 I get "error: pathspec 'r3_8_2' did not match any file(s) known to git"

Any help please...

**Comment by hoffie** [comment]
Hmm, when exactly did you run this? it works for me now (and I think it should have been working since right before the announcement was posted).

Can you post the output of `sudo git remote -v` and `sudo git status`?

Note: We've been providing Debian/Ubuntu packages for some time now which saves you the hassle of recompiling. See https://github.com/jamulussoftware/jamulus/releases/tag/r3_8_2 (Ctrl+F headless)

Note 2: Even when using git you should not need root (sudo) to download and/or compile. In fact, one usually avoids running anything as root if not absolutely necessary. I added my commands with `sudo` above anyway as the owner of your Jamulus repo will likely be `root` right now, so if you wanted to change that, the owner would have to be changed as well (or you would have to clone to a different location).

**Comment by NickVeit** [comment]
Thanks Hoffie

I still get the error on 3_8_2. I didn't mention I also get a message on "sudo git pull" which I considered irrelevant as 3_8_1 works...not sure:
You are not currently on a branch.
Please specify which branch you want to merge with.
See git-pull(1) for details.
    git pull <remote> <branch>

sudo git remote -v reports:
origin  https://github.com/corrados/jamulus.git (fetch)
origin  https://github.com/corrados/jamulus.git (push)

sudo git status reports:
HEAD detached at r3_8_1
nothing to commit, working tree clean

I don't know how to use packages - my knowledge is zero. I just do what I've been told!

I acknowledge the sudo good-practice advice.

In simple terms, what should I type to upgrade. Please note I have specific folders and batch files used to set the IP address with the DNS host so don't want to upset that.

**Reply by hoffie** [reply]
>  didn't mention I also get a message on "sudo git pull" which I considered irrelevant as 3_8_1 works...not sure:
You are not currently on a branch.

Ah, so it looks like the `pull` did not work at all. I think the easiest solution would be the following:

```
sudo git fetch --tags
sudo git checkout r3_8_2
# continue as you're used to
```

In other words, use `git fetch --tags` instead of `git pull`. This is not a Jamulus-side change, btw. It's more likely that either you previously had fresh clones (where a failed pull would not matter) or you may have had a previous version of git which behaved differently in such cases.

Changing to packages would still be beneficial in the long run, but it would probably imply updating your existing scripts to use new paths, etc. So if you want to avoid that right now, that's understandable :)

**Reply by jujudusud** [reply]
@hoffie : github.com/corrados will not work anymore.

**Comment by NickVeit** [comment]
Thanks. Same error still on the checkout.

sudo git fetch --tags reported:

remote: Enumerating objects: 110, done.
remote: Counting objects: 100% (90/90), done.
remote: Total 110 (delta 89), reused 90 (delta 89), pack-reused 20
Receiving objects: 100% (110/110), 32.83 KiB | 16.41 MiB/s, done.
Resolving deltas: 100% (89/89), completed with 48 local objects.
From https://github.com/corrados/jamulus
 * [new tag]           extlogging        -> extlogging
 ! [rejected]          latest            -> latest  (would clobber existing tag)
 * [new tag]           original_contribution_sequence_number_softins -> original_contribution_sequence_number_softi
ns
 * [new tag]           r3.8.0devNightly1 -> r3.8.0devNightly1
 * [new tag]           r3.8.0devNightly2 -> r3.8.0devNightly2
 * [new tag]           r3.8.1devNightly1 -> r3.8.1devNightly1
 * [new tag]           r3_7_0devNightly1 -> r3_7_0devNightly1
 * [new tag]           tests_for_pan_tag -> tests_for_pan_tag

**Reply by hoffie** [reply]
@ann0see just pointed out that you are using an outdated repository address (corrados/jamulus). While this has worked for some time, it no longer does¹. You'll either have to update that or really go the package-based route: https://jamulus.io/wiki/Running-a-Server#running-a-headless-linux-server

**EDIT BY @ann0see**: for everyone reading this, please try without typing sudo first.

So e.g. `sudo git remote remove origin` would be `git remote add origin`.

To update your repo, the following steps will be necessary (only this time):
```
sudo git remote remove origin
sudo git remote add origin https://github.com/jamulussoftware/jamulus
```

Then, continue as mentioned above:
```
sudo git fetch --tags
sudo git checkout r3_8_2
# ...
```

¹ corrados now has its own fork of the new upstream repository here at jamulussoftware/jamulus.

**Reply by NickVeit** [reply]
hoffie and jujudusud - very much appreciated guys. It now works and I understand fully why it didn't this time although I could never have fixed it without the explicit commands you provided.

Updates are few so I'd like to leave things alone for now but will look into packages and the uncertain implications for me at some point.

All the best, Nick

**Reply by ann0see** [reply]
@hoffie could you please update your answer, post it here, mark it and link it from https://github.com/jamulussoftware/jamulus/discussions/2410 ?

**Reply by jujudusud** [reply]
A simple `$ git remote set-url origin https://github.com/jamulussoftware/jamulus.git` is sufficient.

**Comment by ann0see** [comment]
I think you should use the updated version of the script you’re using. 

https://worldjam.vip/jamulus/setup-jamulus.sh might work.

**Reply by NickVeit** [reply]
I'm not using such a script. My setup predates that.

**Comment by jujudusud** [comment]
From my point of view, you should move all the stuff present in `/usr/local/bin/llcon-jamulus` to your `/home/"your-user"/old-jamulus/` to save the old files and refer to it.
After that you just follow the compiling.md file to do that the right way. It is here : [COMPILING.md](https://github.com/jamulussoftware/jamulus/blob/master/COMPILING.md)

**Reply by NickVeit** [reply]
I know nothing about Ubuntu or Unix and have no interest in learning another technology. I got it all working ages ago following the advice at the time. Given the infrequency of updates, my lack of knowledge and the small customisation I have, it seems what might be recommended to others is not necessary for me at this time. I am now pointing at the correct repository and suspect I'll be able to upgrade in 5 minutes again.

**Comment by ann0see** [comment, marked as answer]
*Originally posted by @hoffie in https://github.com/jamulussoftware/jamulus/discussions/2416#discussioncomment-2216726:* 

@ann0see just pointed out that you are using an outdated repository address (corrados/jamulus). While this has worked for some time, it no longer does¹. You'll either have to update that or really go the package-based route: https://jamulus.io/wiki/Running-a-Server#running-a-headless-linux-server

**EDIT BY @ann0see**: for everyone reading this, please try without typing sudo first.

So e.g. `sudo git remote remove origin` would be `git remote add origin`.

To update your repo, the following steps will be necessary (only this time):
```
sudo git remote remove origin
sudo git remote add origin https://github.com/jamulussoftware/jamulus
```

Then, continue as mentioned above:
```
sudo git fetch --tags
sudo git checkout r3_8_2
# ...
```

¹ corrados now has its own fork of the new upstream repository here at jamulussoftware/jamulus.

*Comment/improvement by @jujudusud:*

> A simple `$ git remote set-url origin https://github.com/jamulussoftware/jamulus.git` is sufficient.
